spring-integration-http/src/main/java/org/springframework/integration/http/dsl/HttpMessageHandlerSpec.java

```diff
@@ -0,0 +1,368 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.http.dsl;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.integration.dsl.ComponentsRegistration;
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.http.outbound.HttpRequestExecutingMessageHandler;
+import org.springframework.integration.http.support.DefaultHttpHeaderMapper;
+import org.springframework.integration.mapping.HeaderMapper;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+import org.springframework.web.client.ResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * The {@link MessageHandlerSpec} implementation for the {@link HttpRequestExecutingMessageHandler}.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ *
+ * @see HttpRequestExecutingMessageHandler
+ */
+public class HttpMessageHandlerSpec
+		extends MessageHandlerSpec<HttpMessageHandlerSpec, HttpRequestExecutingMessageHandler>
+		implements ComponentsRegistration {
+
+	private final RestTemplate restTemplate;
+
+	private final Map<String, Expression> uriVariableExpressions = new HashMap<>();
+
+	private HeaderMapper<HttpHeaders> headerMapper = DefaultHttpHeaderMapper.outboundMapper();
+
+	private boolean headerMapperExplicitlySet;
+
+	HttpMessageHandlerSpec(URI uri, RestTemplate restTemplate) {
+		this(new ValueExpression<>(uri), restTemplate);
+	}
+
+	HttpMessageHandlerSpec(String uri, RestTemplate restTemplate) {
+		this(new LiteralExpression(uri), restTemplate);
+	}
+
+	HttpMessageHandlerSpec(Expression uriExpression, RestTemplate restTemplate) {
+		this.target = new HttpRequestExecutingMessageHandler(uriExpression, restTemplate);
+		this.target.setUriVariableExpressions(this.uriVariableExpressions);
+		this.target.setHeaderMapper(this.headerMapper);
+		this.restTemplate = restTemplate;
+	}
+
+	HttpMessageHandlerSpec expectReply(boolean expectReply) {
```

Comment:
It isn't `public` because there is no reason to expose this option to the API, since we can simply restrict an access via `outboundChannelAdapter()/outboundGateway()` factory methods.

Does it make sense for you?